libs/isograph-react/src/EntrypointReader.tsx

```diff
@@ -1,23 +1,27 @@
-import * as React from 'react';
+import * as React from "react";
 import {
   ExtractReadFromStore,
   IsographEntrypoint,
   type FragmentReference,
   useResult,
-} from './index';
+} from "./index";
 
 export function EntrypointReader<
   TProps extends Record<any, any>,
   TEntrypoint extends IsographEntrypoint<any, React.FC<TProps>>,
->(props: {
-  queryReference: FragmentReference<
-    ExtractReadFromStore<TEntrypoint>,
-    React.FC<TProps>
-  >;
-  // TODO make additionalProps optional if TProps extends Record<string, never>
-  // perhaps with a type overload
-  additionalProps: TProps;
-}): React.ReactNode {
+>(
+  props: {
+    queryReference: FragmentReference<
+      ExtractReadFromStore<TEntrypoint>,
+      React.FC<TProps>
+    >;
+    additionalProps?: TProps;
+  } & (TProps extends Record<string, never>
+    ? {}
+    : {
+        additionalProps: unknown;
+      }),
```

Comment:
the idea here is that `{ prop?: Whatever } & { prop: unknown }` is basically equivalent of `{ prop: Whatever }`. So by creating an intersection like this we can easily~ make some "preceding" optional properties required